[CI] RestEsqlIT testForceSleepsProfile {ASYNC} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6551 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/faiy5ummuues2)
- [elasticsearch-periodic-platform-support #6516 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/hiugvpmtjy2fg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.RestEsqlIT.testForceSleepsProfile {ASYNC}" -Dtests.seed=EE63D76CE1F08C05 -Dtests.locale=ia -Dtests.timezone=America/Montevideo -Druntime.java=23
```

**Applicable branches:**
8.18

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.RestEsqlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testForceSleepsProfile%20%7BASYNC%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
      counts: a map containing
exchange empty: expected <1> but was <2>
       first: a list containing
             0: a map containing
          reason: "exchange empty"
    sleep_millis: <1740209910016L>
     wake_millis: <1740209911146L>
             1: <unexpected> but was <{reason=exchange empty, sleep_millis=1740209911388, wake_millis=1740209911507}>
        last: a list containing
             0: a map containing
          reason: "exchange empty"
    sleep_millis: <1740209910016L>
     wake_millis: <1740209911146L>
             1: <unexpected> but was <{reason=exchange empty, sleep_millis=1740209911388, wake_millis=1740209911507}>
```

**Issue Reasons:**
- [8.18] 2 failures in test testForceSleepsProfile {ASYNC} (0.9% fail rate in 226 executions)
- [8.18] 2 failures in step rhel-7_platform-support-unix (33.3% fail rate in 6 executions)
- [8.18] 2 failures in pipeline elasticsearch-periodic-platform-support (33.3% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java
+++ b/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java
@@ -493,26 +493,16 @@ public class RestEsqlIT extends RestEsqlTestCase {
             if (operators.contains("LuceneSourceOperator")) {
                 assertMap(sleeps, matchesMap().entry("counts", Map.of()).entry("first", List.of()).entry("last", List.of()));
             } else if (operators.contains("ExchangeSourceOperator")) {
-                if (operators.contains("ExchangeSinkOperator")) {
-                    assertMap(sleeps, matchesMap().entry("counts", matchesMap().entry("exchange empty", greaterThan(0))).extraOk());
-                    @SuppressWarnings("unchecked")
-                    List<Map<String, Object>> first = (List<Map<String, Object>>) sleeps.get("first");
-                    for (Map<String, Object> s : first) {
-                        assertMap(s, sleepMatcher);
-                    }
-                    @SuppressWarnings("unchecked")
-                    List<Map<String, Object>> last = (List<Map<String, Object>>) sleeps.get("last");
-                    for (Map<String, Object> s : last) {
-                        assertMap(s, sleepMatcher);
-                    }
-
-                } else {
-                    assertMap(
-                        sleeps,
-                        matchesMap().entry("counts", matchesMap().entry("exchange empty", 1))
-                            .entry("first", List.of(sleepMatcher))
-                            .entry("last", List.of(sleepMatcher))
-                    );
+                assertMap(sleeps, matchesMap().entry("counts", matchesMap().entry("exchange empty", greaterThan(0))).extraOk());
+                @SuppressWarnings("unchecked")
+                List<Map<String, Object>> first = (List<Map<String, Object>>) sleeps.get("first");
+                for (Map<String, Object> s : first) {
+                    assertMap(s, sleepMatcher);
+                }
+                @SuppressWarnings("unchecked")
+                List<Map<String, Object>> last = (List<Map<String, Object>>) sleeps.get("last");
+                for (Map<String, Object> s : last) {
+                    assertMap(s, sleepMatcher);
                 }
             } else {
                 fail("unknown signature: " + operators);

